### PR TITLE
ResourceLinkRelation

### PR DESCRIPTION
**Pull request by StephenCote**
Set of changes to add the ResourceLinkRelation feather, model, and formatter.

### CODE REVIEW

**Review by jrogelstad — COMMENTED** [review]
See comments....

File: client/components/resource-link-relation.js [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+
+    This program is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU Affero General Public License for more details.
+
+    You should have received a copy of the GNU Affero General Public License
+    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+*/
+/*jslint this, browser*/
```
[the remark this answers is not in the document]
Comment (StephenCote) [reply client/components/resource-link-relation.js:18]: I have it working via the console now.

File: client/core.js [hunk 2]
```diff
@@ -1393,6 +1394,38 @@ f.hashCode = function (s) {
     );
 };
 
+/// given a data: url, exchange it for a blob url which caches the data to the client to be referenced by the blob id
+///
+function toBlobUrl(dataURI){
+    /// https://stackoverflow.com/questions/9388412/data-uri-to-object-url-with-createobjecturl-in-chrome-ff
+    var mime = dataURI.split(',')[0].split(':')[1].split(';')[0];
+    /// TODO: Replace with Buffer.from(dataURI.split(',')[1],"base64");
```
[the remark this answers is not in the document]
Comment (StephenCote) [reply client/core.js:1402]: Removed as it's unused at the moment.

File: client/core.js [hunk 3]
```diff
@@ -1393,6 +1394,38 @@ f.hashCode = function (s) {
...
+
+f.types.resourceLink = {};
+f.types.resourceLink.tableData = function(obj){
+
+    let dat = obj.value.data;
+    let ico = dat.icon();
+    let rec = dat.resource();
+    let lbl = dat.label();
+
+    let label = (lbl ? lbl : (ico ? "" : rec));
```
[the remark this answers is not in the document]
Comment (StephenCote) [reply client/core.js:1419]: Updated for linting

File: client/components/resource-link-relation.js [hunk 4]
```diff
@@ -0,0 +1,173 @@
...
+                            target: "_blank"
+                        }, [
+                            m("i", {
+                                class: (
+                                    "material-icons " +
+                                    (icon || "") +
+                                    " fb-menu-list-icon"
+                                )
+                            }, "resource")
+                        ], (label || resource))
```
[the remark this answers is not in the document]
Comment (StephenCote) [reply client/components/resource-link-relation.js:87]: I added a class to truncate, but you'll want to review the syntax I used.

File: client/components/resource-link-relation.js [hunk 5]
```diff
@@ -0,0 +1,173 @@
...
+                            href: resource,
+                            target: "_blank"
+                        }, [
+                            m("i", {
+                                class: (
+                                    "material-icons " +
+                                    (icon || "") +
+                                    " fb-menu-list-icon"
+                                )
+                            }, "resource")
```
[the remark this answers is not in the document]
Comment (StephenCote) [reply client/components/resource-link-relation.js:86]: FIxed

File: client/components/resource-link-relation.js [hunk 6]
```diff
@@ -0,0 +1,173 @@
...
+                        ], (label || resource))
+                );
+            }
+
```
[the remark this answers is not in the document]
Comment (StephenCote) [reply client/components/resource-link-relation.js:90]: Fixed

File: scripts/feathers.json [hunk 7]
```diff
@@ -624,6 +624,43 @@
             }
         }
     },
+ 
+    {
```
[the remark this answers is not in the document]
Comment (StephenCote) [reply scripts/feathers.json:628]: Moved

**Pull request merged**